extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -52,15 +69,41 @@ class AmpYoutube extends AMP.BaseElement {
     this.applyFillContent(iframe);
     iframe.width = width;
     iframe.height = height;
+
     this.element.appendChild(iframe);
-    /** @private {?Element} */
+    /** @private {!Element} */
     this.iframe_ = iframe;
-    return loadPromise(iframe);
+    this.iframe_.id = this.frameId_;
+
+    return loadPromise(iframe).then(() => {
+      // Make sure the YT player is ready for this. For some reason YT player
+      // would send couple of messages but then stop. Waiting for a bit before
+      // sending the 'listening' event seems to fix that and allow YT Player
+      // to send messages continuously.
+      setTimeout(() => {
+        // Keep track of the player state.
+        listen(iframe, 'infoDelivery', data => {
+          if (data.info && data.info.playerState !== undefined) {
+            self.playerState_ = data.info.playerState;
```

Comment:
What do we get out of these messages? Is it critical to know whether or not the video is playing before pausing?